generators/client/templates/angular/src/main/webapp/app/shared/model/_request-util.ts

```diff
@@ -1,18 +1,18 @@
-import { URLSearchParams, BaseRequestOptions } from '@angular/http';
+import { HttpParams } from '@angular/common/http';
 
-export const createRequestOption = (req?: any): BaseRequestOptions => {
-    const options: BaseRequestOptions = new BaseRequestOptions();
+export const createRequestOption = (req?: any, sort?: any): HttpParams => {
+    let options: HttpParams = new HttpParams();
     if (req) {
-        const params: URLSearchParams = new URLSearchParams();
-        params.set('page', req.page);
-        params.set('size', req.size);
-        if (req.sort) {
-            params.paramsMap.set('sort', req.sort);
+        for (const key in req) {
```

Comment:
dont use for in as its heavy and brings in generator runtime, use
```
Object.keys(req).forEach(key =>  options.set(key, req[key]));
```
  hasOwnProperty check not required for Object.keys